packages/gensx/src/dev-server.ts

```diff
@@ -0,0 +1,1192 @@
+          this.executionsMap.set(executionId, execution);
+
+          // Handle different response types
+          if (
+            result &&
+            typeof result === "object" &&
+            Symbol.asyncIterator in result
+          ) {
+            // Handle streaming responses
+            return this.handleStreamingResponse(
+              c,
+              result as AsyncIterable<unknown>,
+            );
+          }
+
+          // Handle regular JSON response
+          return c.json({
+            status: "ok",
+            data: {
+              executionId,
+              executionStatus: "completed",
+              output: result,
+            },
+          });
+        } catch (error) {
+          // For validation errors, don't create an execution record
+          if (error instanceof BadRequestError) {
+            console.error(
+              `❌ Validation error in workflow '${workflowName}':`,
+              error.message,
+            );
+            return c.json(
+              {
+                status: "error",
+                error: error.message,
+              },
+              400,
+            );
+          }
+
+          console.error(
+            `❌ Error executing workflow '${workflowName}':`,
+            error,
+          );
+
+          // For other errors, proceed with server error
+          return c.json(
+            {
+              status: "error",
+              error: error instanceof Error ? error.message : String(error),
+            },
+            500,
+          );
+        }
+      },
+    );
+
+    // UI for testing workflows
+    this.app.get("/openapi.json", (c) => {
+      return c.json(this.generateOpenApiSpec());
+    });
+
+    this.app.get("/swagger-ui", (c) => {
+      const html = this.generateSwaggerUI();
+      return c.html(html);
+    });
+  }
+
+  /**
+   * Handle streaming responses
+   */
+  private handleStreamingResponse(
+    _c: Context,
+    streamResult: AsyncIterable<unknown>,
+  ) {
+    const stream = new ReadableStream({
+      async start(controller) {
+        try {
+          for await (const chunk of streamResult) {
+            // Stringify the chunk if it's not already a string
+            const chunkStr =
+              typeof chunk === "string" ? chunk : JSON.stringify(chunk) + "\n";
+            controller.enqueue(new TextEncoder().encode(chunkStr));
+          }
+
+          // Close the stream
+          controller.close();
+        } catch (error) {
+          console.error("❌ Error in streaming response:", error);
+          controller.error(error);
+        }
+      },
+    });
+
+    return new Response(stream, {
+      headers: {
+        "Content-Type": "application/stream",
+        "Transfer-Encoding": "chunked",
+      },
+    });
+  }
+
+  /**
+   * Generate OpenAPI specification dynamically based on server configuration
+   */
+  private generateOpenApiSpec(): Record<string, unknown> {
```

Comment:
We should port this over to production soon.

Reply:
fair, i'll add it to my list

Reply:
created an issue and assigned it to myself https://github.com/gensx-inc/gensx-console/issues/149